AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Internal/URLComponents+sigv4Encoding.swift

```diff
@@ -22,15 +22,29 @@ extension URLComponents {
         var sigV4UnreservedCharacters = CharacterSet(charactersIn: "A" ... "Z")
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "a" ... "z"))
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "0" ... "9"))
-        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~.%=&"))
+        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~."))
         return sigV4UnreservedCharacters
     }()
 
-    mutating func percentEncodeQueryBySigV4Rules() {
-        guard let percentEncodedQuery = self.percentEncodedQuery else {
+    private func encodeQueryParamValueBySigV4Rules(_ value: String) -> String {
+        /// removingPercentEncoding returns `nil` if called on a value
+        /// that hasn't been prior encoded
+        let unencoded = value.removingPercentEncoding ?? value
+
+        /// apply sigv4 encoding
+        return unencoded.addingPercentEncoding(
+            withAllowedCharacters: Self.sigV4UnreservedCharacters) ?? value
+    }
+
+    mutating func queryItemsEncodedPerSigV4Rules(_ queryItems: [String: String]?) {
+        guard let queryItems = queryItems else {
             return
         }
-        self.percentEncodedQuery = percentEncodedQuery.addingPercentEncoding(
-            withAllowedCharacters: Self.sigV4UnreservedCharacters)
+        percentEncodedQuery = queryItems.map { (name, value) -> String in
+            let encodedName = encodeQueryParamValueBySigV4Rules(name)
+            let encodedValue = encodeQueryParamValueBySigV4Rules(value)
```

Comment:
Such nit, very picky: if it wouldn't be a breaking change...you're encoding an arbitrary string value, not just a query parameter value. More specifically, you're encoding both a query "field" and "value". Consider name change to something like `encodeQueryParamItemBySignV4Rules(_:)`. Or even `encodeBySigV4Rules(_:)`?